### PR TITLE
Added some potential way to test

### PR DESCRIPTION
**Pull request by javiersuweijie**
@MaxGraey, what do you think of testing the library using nodejs and mocha. You can try `yarn run build:test && yarn run test` to run them.

### CODE REVIEW

**Comment by MaxGraey** [comment]
@javiersuweijie Thanks a lot!

File: assembly/globals.ts [hunk 1]
```diff
@@ -83,7 +83,7 @@ export function __umulh64(u: u64, v: u64): u64 {
 
 @global
 export function __umulq64(_res: usize, u: u64, v: u64): void {
-  var u1, v1, w0, w1, t;
+  var u1 : u64 , v1 : u64, w0 : u64, w1 : u64, t : u64;
```
Comment (MaxGraey) [inline assembly/globals.ts]: Remove plz spaces between itentifier and `:` like `u1: u64` for convention with other formatting style. Later I'll add tslint with according rules.

File: assembly/integer/u256.ts [hunk 2]
```diff
@@ -188,7 +188,7 @@ export class u256 {
   }
 
   @inline @operator('+')
-  static add(a: u256, b: u256) {
+  static add(a: u256, b: u256) : u256 {
```
Comment (MaxGraey) [inline assembly/integer/u256.ts]: Same. Remove plz space before `:`

File: test/helloword.spec.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
+import { expect } from 'chai';
+import { describe, it } from "mocha";
+import * as fs from 'fs';
+
+describe('Sanity check for WebAssembly', () => {
+    it("should be available", () => {
```
Comment (MaxGraey) [inline test/helloword.spec.ts]: Make indent equal 2 spaces plz

File: test/helloword.spec.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        let p = instance.plus(1,1);
+        expect(buffer[p]).to.be.eq(2);
+    })
+    it("should add 100 plus 100", () => {
+        let p = instance.plus(100,100);
+        expect(buffer[p]).to.be.eq(200);
+    })
+    it("should add 1000000 plus 1000000", () => {
+        let p = instance.plus(1000000,1000000);
+        for (var i = 0; i < 32; i++) {
```
Comment (MaxGraey) [inline test/helloword.spec.ts]: It seems this part great candidate for making general util which check u128 result.

File: test/u128/u128.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import 'allocator/arena';
+import { u128 } from '../../assembly/integer/u128';
+
+
+export function plus(a : u32, b : u32) : usize {
+    // return changetype<usize>(u128.fromI32(a) + u128.fromI32(b));
+    return changetype<usize>(u128.from<u32>(a) + u128.from<u32>(b));
```
Comment (MaxGraey) [inline test/u128/u128.ts]: You can simplify this to:
```ts
return changetype<usize>(u128.from(a) + u128.from(b));
```
Generic type should easily resolved for that case

File: tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,59 @@
+{
+  "compilerOptions": {
+    /* Basic Options */
+    "target": "es5",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
```
Comment (MaxGraey) [inline tsconfig.json]: I think we can use "esnext" for current day and latest node.

File: test/helloword.spec.ts [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+describe('Basic u128 operations', () => {
+    var file = fs.readFileSync("test/build/u128.wasm");
+    var memory = new WebAssembly.Memory({initial: 1});
+    var buffer = new Uint8Array(memory.buffer);
+    var instance : any;
+
+    var imports = {
+        env: {
+            abort: () => {},
```
Comment (MaxGraey) [inline test/helloword.spec.ts:20]: Add plz proper abort function.
```ts
abort(msg, file, line, column) {
   console.error(`abort called at ${ file } (${ line }:${ column })`);
}
```

**Comment by javiersuweijie** [comment]
@MaxGraey i made the changes. will try to fix the failing builds in another pull request

**Pull request merged**